library/fortimgr_jsonrpc_request.py

```diff
@@ -0,0 +1,1508 @@
+
+        :return: The JSON response from the request to save the session.
+        """
+        body = {"method": "exec", "params": [{"url": self.wsp_url + "commit"}], "session": self.session}
+        response = self.make_request(body)
+
+        return response.json()
+
+    def unlock(self):
+        """
+        The unlock method is used to lock the ADOM to enable configurations to be sent to the FortiManager when it has
+        workspace mode enabled.
+
+        :return: The JSON response from the request to unlock the session.
+        """
+        body = {"method": "exec", "params": [{"url": self.wsp_url + "unlock"}], "session": self.session}
+        response = self.make_request(body)
+
+        return response.json()
+
+    def update_config(self, update_config):
+        """
+        This method is used to submit a configuration update request to the FortiManager. Only the object configuration
+        details need to be provided; all other parameters that make up the API request body will be handled by the
+        method. Only fields that need to be updated are required to be in the "update_config" variable (EX: updating
+        the comment for an address group only needs the "name" and "comment" fields in the configuration dictionary).
+        When including a field in the configuration update, ensure that all items are included for the desired end-state
+        (EX: adding address to an address group that already has ["svr01", "svr02"] should include all three
+        addresses in the "member" list, ["svr01", "svr02", "svr03"]. If you want to remove part of an item's
+        configuration, this method should be used, and the item to be removed should be left off the respective list
+        (EX: removing an address from an address group that has ["svr01", "svr02", "svr03"] should have a "member" list
+        like ["svr01", "svr02"] with the final state of the address group containing only svr01 and svr02).
+
+        :param update_config: Type list.
+                           The "data" portion of the configuration to be submitted to the FortiManager.
+        :return: The response from the API request to add the configuration.
+        """
+        body = {"method": "update", "params": [{"url": self.obj_url, "data": update_config, "session": self.session}]}
+        response = self.make_request(body)
+
+        return response
+
+    def _escape_params_url(self, url):
+        """
+        This private method is used to escape slash ("/") characters from a url string to be provided as a json-rpc request params.
+        Slash characters are escaped by prefixing with a backslash ("\").
+        If url is None, None is returned.
+
+        :param url: Type str.
+                        The url string to process.
+        :return: The url string with slash characters escaped with a backslash ("\") or None if url is None.
+        """
+        if url is not None:
+            return url.replace('/', '\\/')
```

Comment:
needs to include the string typecast

Reply:
other than that, there are a couple of tabs instead of spaces in the files; clean these up and it should be good to merge